version: "3.8"

services:
  database:
      image: mysql:latest
      env_file:
        - .env
      ports:
        - "3306:3306"
      container_name: database
      environment:
        MYSQL_ROOT_PASSWORD: ${MYSQL_PASSWORD}
        MYSQL_DATABASE: ${MYSQL_DATABASE}
        MYSQL_USER: ${MYSQL_USER}
        MYSQL_PASSWORD: ${MYSQL_PASSWORD}
      healthcheck:
        test:
          [
            "CMD",
            "mysqladmin",
            "ping",
            "-h",
            "localhost",
            "-u",
            "root",
            "--password=root",
          ]
        timeout: 20s
        retries: 10
      volumes:
        - ./databaseScript:/docker-entrypoint-initdb.d
      networks:
        - my-network

  app:
    build: .
    env_file:
      - .env
    ports:
      - "8000:8000"
    container_name: app
    depends_on:
      database:
        condition: service_healthy
    environment:
      - DATABASE_URI=mysql://${MYSQL_USER}:${MYSQL_PASSWORD}@database:3306/${MYSQL_DATABASE}
    restart: always
    networks:
      - my-network

networks:
  my-network:
    driver: bridge

